src/NzbDrone.Core/Download/Clients/Tribler/TriblerDownloadClientProxy.cs

```diff
@@ -0,0 +1,137 @@
+
+            try
+            {
+                response = _httpClient.Execute(httpRequest);
+            }
+            catch (HttpException ex)
+            {
+                if (ex.Response.StatusCode == System.Net.HttpStatusCode.Unauthorized)
+                {
+                    throw new DownloadClientAuthenticationException("Unauthorized - AuthToken is invalid", ex);
+                }
+
+                throw new DownloadClientUnavailableException("Unable to connect to Tribler. Status Code: {0}", ex.Response.StatusCode, ex);
+            }
+
+            return Json.Deserialize<T>(response.Content);
+        }
+
+        public GetTriblerSettingsResponse GetConfig(TriblerDownloadSettings settings)
+        {
+            var configRequest = getRequestBuilder(settings, "settings");
+            return ProcessRequest<GetTriblerSettingsResponse>(configRequest);
+        }
+
+        public ICollection<File> GetDownloadFiles(TriblerDownloadSettings settings, Download downloadItem)
+        {
+            var filesRequest = getRequestBuilder(settings, "downloads/" + downloadItem.Infohash + "/files");
+            return ProcessRequest<GetFilesResponse>(filesRequest).Files;
+        }
+
+        public ICollection<Download> GetDownloads(TriblerDownloadSettings settings)
+        {
+            var downloadRequest = getRequestBuilder(settings, "downloads");
+            var downloads = ProcessRequest<DownloadsResponse>(downloadRequest);
+            return downloads.Downloads;
+        }
+
+        public void RemoveDownload(TriblerDownloadSettings settings, DownloadClientItem item, bool deleteData)
+        {
+            var deleteDownloadRequestObject = new RemoveDownloadRequest
+            {
+                RemoveData = deleteData
+            };
+
+            var deleteRequestBuilder = getRequestBuilder(settings, "downloads/" + item.DownloadId.ToLower());
+            deleteRequestBuilder.Method = System.Net.Http.HttpMethod.Delete;
+
+            // manually set content of delete request.
+            var deleteRequest = deleteRequestBuilder.Build();
+            deleteRequest.SetContent(Json.ToJson(deleteDownloadRequestObject));
+
+            ProcessRequest<DeleteDownloadResponse>(deleteRequest);
+        }
+
+        public string AddFromMagnetLink(TriblerDownloadSettings settings, AddDownloadRequest downloadRequest)
+        {
+            var addDownloadRequestBuilder = getRequestBuilder(settings, "downloads");
+            addDownloadRequestBuilder.Method = System.Net.Http.HttpMethod.Put;
+
+            // manually set content of download request.
+            var addDownloadRequest = addDownloadRequestBuilder.Build();
+            addDownloadRequest.SetContent(Json.ToJson(downloadRequest));
+
+            var infoHashAsString = ProcessRequest<AddDownloadResponse>(addDownloadRequest).Infohash;
+
+            // run hash through InfoHash class to ensure the correct casing.
```

Comment:
Why is the casing from Tribler incorrect?
If it's incorrect here won't it also be incorrect when returned from the API when getting the queue?

If Tribler is giving different casing than expected then Sonarr will lose track of it and trying to massage it here won't add any value and should be removed.

Reply:
It is a really good question, i will find some examples & try it again without the manual case fixing.
The id's returned from tribler are lowercase, eg: f12a0d1bcd77cb3d9b77029bc18374c6c58626d7

Reply:
Not suer what casing we track things under off hand, but we can just fix the case instead of running it through `InfoHash`, the same would be needed for getting torrents from the queue.

Reply:
I have been running with the info hash conversion disabled for a while and it does not seem to cause issues. So either it's fixed by updating tribler, or rebasing with Sonar..